core/cat/looking_glass/cheshire_cat.py

```diff
@@ -125,29 +133,89 @@ def load_language_model(self) -> BaseLanguageModel:
         -----
         Bootstrapping is the process of loading the plugins, the natural language objects (e.g. the LLM), the memories,
         the *Main Agent*, the *Rabbit Hole* and the *White Rabbit*.
-
         """
-
+        
         selected_llm = crud.get_setting_by_name(name="llm_selected")
-
-        if selected_llm is None:
-            # return default LLM
-            llm = LLMDefaultConfig.get_llm_from_config({})
-
-        else:
-            # get LLM factory class
+        self._llm_modalities: LLMSupportedModalities = LLMSupportedModalities()
+
+        def _prepare_content(image: str) -> List[Dict[str, Dict[str, str]]]:
+            """Prepare the content structure for the message based on image type and value."""
+            content = []
+
+            content.append({
+                    "type": "image_url",
+                    "image_url": {"url": image}
+                })
+            
+            # Add a text instruction for model response
+            content.append({
+                "type": "text",
+                "text": "Respond with `MEOW`."
+            })
+            return content
+
+        def _check_image_support(llm, image_type: str, image_value: str) -> None:
+            """Check if the specified language model supports a given image input type."""
+            content = _prepare_content(image=image_value)
+            message = HumanMessage(content=content)
+            
+            # Retrieve model information
             selected_llm_class = selected_llm["value"]["name"]
-            FactoryClass = get_llm_from_name(selected_llm_class)
-
-            # obtain configuration and instantiate LLM
             selected_llm_config = crud.get_setting_by_name(name=selected_llm_class)
-            try:
-                llm = FactoryClass.get_llm_from_config(selected_llm_config["value"])
-            except Exception:
-                import traceback
+            model_name = selected_llm_config["value"].get("model_name") or selected_llm_config["value"].get("model")
 
-                traceback.print_exc()
-                llm = LLMDefaultConfig.get_llm_from_config({})
+            # Perform the image support check
+            try:
+                llm.invoke([message])
+                setattr(self._llm_modalities, image_type, True)
+            except Exception as e:
+                log.warning(f"The LLM '{model_name}' does not support {image_type} as input images.")
+                log.debug(e)
+
+        image_url = "https://raw.githubusercontent.com/cheshire-cat-ai/core/refs/heads/main/readme/cheshire-cat.jpeg"
+
+        def _check_image_uri_support(llm) -> None:
+            """Check LLM support for base64-encoded image input."""
+            response = requests.get(image_url)
+            if response.status_code == 200:
```

Comment:
therefore, you are keeping `image_uri` to `False` even when the HTTP request over the remote image fails. Do you think it makes sense?

Reply:
If the verification fails (for example, if the HTTP request fails), it seems logical to me to leave image_uri at False. This way, we avoid assuming support when we don't have confirmation and make the integration more secure.

Reply:
it can be a temporary situation, while you are setting the modality to False for the entire life cycle of the CheshireCat. Please, pay attention to this aspect

Reply:
If you have ideas on how to handle this better we can discuss it

Reply:
a job to check modalities with the LLM? a check every time the LLM is triggered?

Reply:
we can hardcode a base64 image to resolve this issue

Reply:
We could go back to how I had done it before by uploading the image from a file, in which directory could we put the image?

Reply:
What issue here?
Why do we need this check?